frontend/src/js/components/subscription/SubscriptionConfirmation.tsx

```diff
@@ -0,0 +1,129 @@
+// Copyright 2025 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { TaskAlt as TaskAltIcon } from '@mui/icons-material';
+import { Alert, AlertTitle, Button, DialogContent, Divider, IconButton, Typography } from '@mui/material';
+
+import { BaseDialog } from '@northern.tech/common-ui/dialogs/BaseDialog';
+import { cleanUp } from '@northern.tech/store/auth';
+import { ADDONS, AvailableAddon, Plan } from '@northern.tech/store/constants';
+import { getOrganization } from '@northern.tech/store/organizationSlice/selectors';
+
+import { formatPrice } from './utils';
+
+interface SubscriptionConfirmationProps {
+  devices: number;
+  onClose: () => void;
+  orderedAddons: { name: AvailableAddon }[];
+  plan: Plan;
+  price: number;
+}
+export const SubscriptionConfirmation = (props: SubscriptionConfirmationProps) => {
+  const { plan, devices, price, orderedAddons, onClose } = props;
+  const { addons: enabledAddons, plan: currentPlan } = useSelector(getOrganization);
+
+  const [addonList] = useState(orderedAddons.map(addon => addon.name));
+
+  const previousAddonsList = enabledAddons.filter(addon => addon.enabled);
+  const willLogout = addonList.length > previousAddonsList.length || currentPlan !== plan.id;
```

Comment:
I guess that will have to take the addons in question into account as well, to catch a change from one addon to another too..

Reply:
For now we don't allow removing already purchased addons, and I have a feeling that it won't be prioritized any time soon :smiley: , but good catch anyway